File: datasets/rb/HumanEval_98_count_upper.rb
# Given a string s, count the number of uppercase vowels in even indices.
# For example:
# count_upper('aBCdEf') returns 1
# count_upper('abcdefg') returns 0
# count_upper('dBBE') returns 0
def count_upper(s)
  count = 0
  s.each_char.with_index do |char, i|
    count += 1 if char.upcase == char && i.even?
  end
  count
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_count_upper
    candidate = method(:count_upper)
    assert_equal(1, candidate.call("aBCdEf"))
    assert_equal(0, candidate.call("abcdefg"))
    assert_equal(0, candidate.call("dBBE"))
    assert_equal(0, candidate.call("B"))
    assert_equal(1, candidate.call("U"))
    assert_equal(0, candidate.call(""))
    assert_equal(2, candidate.call("EEEE"))
  end
end

File: datasets/rs/HumanEval_98_count_upper.rs
/// Given a string s, count the number of uppercase vowels in even indices.
/// For example:
/// count_upper('aBCdEf') returns 1
/// count_upper('abcdefg') returns 0
/// count_upper('dBBE') returns 0
fn count_upper(s: String) -> isize {
    let mut count = 0;
    for (i, c) in s.chars().enumerate() {
        if i % 2 == 0 && c.is_uppercase() {
            count += 1;
        }
    }
    count

}

fn main() {
    let candidate = count_upper;
    assert_eq!(candidate(String::from("aBCdEf")), 1);
    assert_eq!(candidate(String::from("abcdefg")), 0);
    assert_eq!(candidate(String::from("dBBE")), 0);
    assert_eq!(candidate(String::from("B")), 0);
    assert_eq!(candidate(String::from("U")), 1);
    assert_eq!(candidate(String::from("")), 0);
    assert_eq!(candidate(String::from("EEEE")), 2);
}
